plugins/org.eclipse.fordiac.ide.fbtypeeditor.network/src/org/eclipse/fordiac/ide/fbtypeeditor/network/viewer/AbstractFbNetworkInstanceViewer.java

```diff
@@ -100,12 +100,23 @@ protected void initializeGraphicalViewer() {
 
 	@Override
 	public void setInput(final IEditorInput input) {
-		if (input instanceof final CompositeAndSubAppInstanceViewerInput untypedInput) {
-			fbNetworkElement = untypedInput.getContent();
+		if (!(input instanceof final CompositeAndSubAppInstanceViewerInput viewerEI)) {
+			throw new IllegalArgumentException(
+					"Network viewers only accept CompositeAndSubAppInstanceViewerInput as valid inputs!"); //$NON-NLS-1$
+		}
+		final CompositeAndSubAppInstanceViewerInput currentEditorInput = (CompositeAndSubAppInstanceViewerInput) getEditorInput();
+		if (currentEditorInput != null && currentEditorInput.getContent() != viewerEI.getContent()) {
+			throw new IllegalArgumentException(
+					"Editor input with new content given to network viewer. This is currently not supported!"); //$NON-NLS-1$
+		}
```

Comment:
I believe this check is now redundant with `DiagramEditor.setInput(IEditorInput)`?

Reply:
fixed as above.